modules/apps/analytics/analytics-settings-rest-client/src/main/java/com/liferay/analytics/settings/rest/client/pagination/Page.java

```diff
@@ -192,44 +190,59 @@ protected void setField(
 			}
 			else if (Objects.equals(jsonParserFieldName, "facets")) {
 				if (jsonParserFieldValue != null) {
-					page.setFacets(
-						Stream.of(
-							toStrings((Object[])jsonParserFieldValue)
-						).map(
-							this::parseToMap
-						).map(
-							facets -> new Facet(
-								(String)facets.get("facetCriteria"),
-								Stream.of(
-									(Object[])facets.get("facetValues")
-								).map(
-									object -> (String)object
-								).map(
-									this::parseToMap
-								).map(
-									facetValues -> new Facet.FacetValue(
-										Integer.valueOf(
-											(String)facetValues.get(
-												"numberOfOccurrences")),
-										(String)facetValues.get("term"))
-								).collect(
-									Collectors.toList()
-								))
-						).collect(
-							Collectors.toList()
-						));
+					List<Facet> facets = new ArrayList<>();
+
+					for (String str :
+							toStrings((Object[])jsonParserFieldValue)) {
+
+						List<Facet.FacetValue> facetValues = new ArrayList<>();
+
+						for (Object obj :
+								(Object[])parseToMap(
```

Comment:
@jeffwu0724 The output here is pretty messy, please avoid chaining on these parseToMap calls in the ftl.